File: 1700. Number of Students Unable to Eat Lunch/countStudents.py
from typing import List
from collections import Counter

class Solution:
  def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
    count = {}
    
    for student in students:
      if student not in count:
        count[student] = 0
      count[student] += 1

    for i in range(len(sandwiches)):
      if count.get(sandwiches[i], 0) == 0:
        return len(sandwiches) - i
      count[sandwiches[i]] -= 1
      
    return 0
  
students = [1, 1]
sandwiches = [0, 1]
solution = Solution()
print('countStudents: ', solution.countStudents(students, sandwiches))

File: 1700. Number of Students Unable to Eat Lunch/countStudents.js
/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
const countStudents = (students, sandwiches) => {
  const count = {};
  for (let i = 0; i < students.length; i++) {
    if (!count[students[i]]) {
      count[students[i]] = 0;
    }
    count[students[i]]++;
  }

  for (let i = 0; i < sandwiches.length; i++) {
    if ((count[sandwiches[i]] || 0) === 0) {
      return sandwiches.length - i;
    }
    count[sandwiches[i]]--;
  }

  return 0;
};

const students = [1, 1];
const sandwiches = [0, 1];
console.log('countStudents: ', countStudents(students, sandwiches))
